mdata/cwr.go

```diff
@@ -1,23 +1,78 @@
 package mdata
 
 import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
 	"time"
 
-	"github.com/grafana/metrictank/mdata/chunk"
 	"github.com/raintank/schema"
 )
 
 // ChunkWriteRequest is a request to write a chunk into a store
+//go:generate msgp
 type ChunkWriteRequest struct {
-	Metric    *AggMetric
+	Callback  func() `msg:"-"`
 	Key       schema.AMKey
-	Chunk     *chunk.Chunk
 	TTL       uint32
-	Span      uint32
+	T0        uint32
+	Data      []byte
 	Timestamp time.Time
 }
 
 // NewChunkWriteRequest creates a new ChunkWriteRequest
-func NewChunkWriteRequest(metric *AggMetric, key schema.AMKey, chunk *chunk.Chunk, ttl, span uint32, ts time.Time) ChunkWriteRequest {
-	return ChunkWriteRequest{metric, key, chunk, ttl, span, ts}
+func NewChunkWriteRequest(callback func(), key schema.AMKey, ttl, t0 uint32, data []byte, ts time.Time) ChunkWriteRequest {
+	return ChunkWriteRequest{callback, key, ttl, t0, data, ts}
+}
+
+// MdWithCwrs is used by the whisper importer utilities to transfer the data
+// from the reader to the writer
+//go:generate msgp
+type MdWithCwrs struct {
```

Comment:
i do not like this struct name at all.    We want a struct to allow us to transport a batch of ChunkWriteRequests and the metadata needed to add an entry index.

I think the the old 'Archive' name was better. But perhaps we can use `ArchiveRequest`